packages/fxa-payments-server/src/components/NewUserEmailForm/index.test.tsx

```diff
@@ -137,23 +138,47 @@ describe('NewUserEmailForm test', () => {
     });
   });
 
+  it('shows error when an account already exists', async () => {
+    let subject;
+    await act(async () => {
+      subject = render(
+        <WrapNewUserEmailForm accountExistsReturnValue={true} />
+      );
+      const firstEmail = subject.getByTestId('new-user-email');
+
+      fireEvent.change(firstEmail, { target: { value: 'valid@email.com' } });
+      fireEvent.blur(firstEmail);
+    });
+
+    expect(subject.queryByTestId('already-have-account-copy')).toBeVisible();
+  });
+
   it('Notifies the user if they already have an account', async () => {
     const checkAccountExists = (userAccount: string) =>
       Promise.resolve({ exists: true });
 
-    emailInputValidationAndAccountCheck(
+    const result = await emailInputValidationAndAccountCheck(
       'foxy@mozilla.com',
       false,
-      (state: string) => state,
-      (value: boolean) => value,
-      (id: string) => id,
-      checkAccountExists
-    ).then((result) => {
-      expect(result).toEqual({
-        value: 'foxy@mozilla.com',
-        valid: true,
-        error: 'new-user-existing-account-sign-in',
-      });
-    });
+      (state: string) => {},
+      (value: boolean) => {},
+      (value: string) => {},
+      checkAccountExists,
+      'example.com/signin',
+      (id: string) => id
+    );
+
+    expect(result.value).toEqual('foxy@mozilla.com');
+    expect(result.valid).toEqual(true);
+    expect(result.error).toMatchObject(
+      <Localized
+        elems={{ a: <a href="example.com/signin" /> }}
+        id="new-user-existing-account-sign-in"
+      >
+        <p data-testid="already-have-account-copy">
+          You already have an account, <a href="example.com/signin">Sign in</a>
+        </p>
+      </Localized>
+    );
```

Comment:
This last assertion for `result.error` is overly specific; is there a better assertion I can make here? Most of the element-based assertions seem to require a `JSX.Element`.